src/Streamly/Internal/Data/Parser/ParserD/Tee.hs

```diff
@@ -358,9 +387,21 @@ shortest (Parser stepL initialL extractL) (Parser stepR initialR _) =
 
     {-# INLINE_LATE initial #-}
     initial = do
-        sL <- initialL
-        sR <- initialR
-        return $ TeePair ([], StepState sL, [], []) ([], StepState sR, [], [])
+        resL <- initialL
+        resR <- initialR
+        return $ case resL of
+            IPartial sl ->
+                case resR of
+                     IPartial sr -> IPartial $ TeePair ([], StepState sl, [], [])
+                                                       ([], StepState sr, [], [])
+                     IDone br -> IDone br
+                     IError err -> IError err
+            IDone bl -> IDone bl
+            IError errL ->
+                case resR of
+                     IPartial _ -> IError errL
+                     IDone br -> IDone br
```

Comment:
Is this the desired behaviour? or should it return `IError errL` ?

Reply:
Yeah, We want the `shortest` to succeed if possible. Does not matter if the other one fails at the same length.